src/background/tab-state.js

```diff
@@ -82,62 +96,108 @@ export default function TabState(initialState, onchange) {
     currentState = newCurrentState;
   };
 
+  /**
+   * Mark a tab as having Hypothesis loaded in it.
+   *
+   * @param {number} tabId
+   */
   this.activateTab = function (tabId) {
-    this.setState(tabId, { state: states.ACTIVE });
+    this.setState(tabId, { state: 'active' });
   };
 
+  /**
+   * Mark a tab as not having Hypothesis loaded.
+   *
+   * @param {number} tabId
+   */
   this.deactivateTab = function (tabId) {
-    this.setState(tabId, { state: states.INACTIVE });
+    this.setState(tabId, { state: 'inactive' });
   };
 
+  /**
+   * Mark a tab as having encountered an error when trying to load Hypothesis into it.
+   *
+   * @param {number} tabId
+   * @param {Error} error
+   */
   this.errorTab = function (tabId, error) {
     this.setState(tabId, {
-      state: states.ERRORED,
+      state: 'errored',
       error: error,
```

Comment:
Out of curiosity: `error` object is skip in the serialisation on the `localstorage`?

Reply:
Yes. The `TabStore.set` method only persists the `state` and `annotationCount` properties. It is possible this could cause problems. I haven't checked.